misc/lldb_disasm.py

```diff
@@ -0,0 +1,196 @@
+#!/usr/bin/env python
+#coding: utf-8
+#
+# Usage: run `command script import -r misc/lldb_disasm.py` on LLDB
+#
+#
+#    (lldb) p iseq
+#    (rb_iseq_t *) $147 = 0x0000000101068400
+#    (lldb) rbdisasm iseq
+#    0000 putspecialobject( 3 )
+#    0002 putnil
+#    0003 defineclass( ID: 0x560b, (rb_iseq_t *)0x1010681d0, 2 )
+#    0007 pop
+#    0008 putspecialobject( 3 )
+#    0010 putnil
+#    0011 defineclass( ID: 0x56eb, (rb_iseq_t *)0x101063b58, 2 )
+#    0015 leave
+
+
+import lldb
+import os
+import shlex
+
+TS_VARIABLE = b'.'[0]
+TS_CALLDATA = b'C'[0]
+TS_CDHASH   = b'H'[0]
+TS_IC       = b'K'[0]
+TS_IVC      = b'A'[0]
+TS_ID       = b'I'[0]
+TS_ISE      = b'T'[0]
+TS_ISEQ     = b'S'[0]
+TS_OFFSET   = b'O'[0]
+TS_VALUE    = b'V'[0]
+TS_LINDEX   = b'L'[0]
+TS_FUNCPTR  = b'F'[0]
+TS_NUM      = b'N'[0]
+TS_BUILTIN  = b'R'[0]
+
+def rb_vm_insn_addr2insn2(target, result, wanted_addr, size):
+    tIntPtr = target.FindFirstType("intptr_t")
+    sizeOfIntPtr = tIntPtr.GetByteSize();
+    addr_of_table = target.FindSymbols("vm_exec_core.insns_address_table")[0].GetSymbol().GetStartAddress().GetLoadAddress(target)
+
+    for insn in range(0, size):
+        addr_in_table = addr_of_table + (insn * sizeOfIntPtr)
+        addr = lldb.SBAddress(addr_in_table, target)
+        machine_insn = target.CreateValueFromAddress("insn", addr, tIntPtr).GetValueAsUnsigned()
+
+        if machine_insn == wanted_addr:
+            return insn
+    return False
+
+def iseq_extract_values(debugger, target, process, result, iseqs, n):
+    tValueP      = target.FindFirstType("VALUE")
+    sizeofValueP = tValueP.GetByteSize()
+    insn = target.CreateValueFromAddress(
+            "i", lldb.SBAddress(iseqs.unsigned + (n * sizeofValueP), target), tValueP)
+    addr         = insn.GetValueAsUnsigned()
+    size         = target.EvaluateExpression('ruby_vminsn_type::VM_INSTRUCTION_SIZE').unsigned
+    orig_insn    = rb_vm_insn_addr2insn2(target, result, addr, size)
+
+    name     = insn_name(target, process, result, orig_insn)
+    length   = insn_len(target, orig_insn)
+    op_types = bytes(insn_op_types(target, process, result, orig_insn), 'utf-8')
+
+    print("%04d %s" % (n, name), file=result, end="")
+
+    if length == 1:
+        print("", file=result)
+        return length
+
+    idx = 0
+    print("(", end="", file=result)
+    for op_type in op_types:
```

Comment:
`for op_type in op_types:` -> `for idx, op_type in enumerate(op_types):` this way you wouldn't have to manually increment `idx`.

https://docs.python.org/3.8/library/functions.html#enumerate